tracpro/baseline/views.py

```diff
@@ -144,8 +155,27 @@ def form_valid(self, form):
                     submitted_on=baseline_datetime,
                     category=u'')
 
+        def form_valid(self, form):
+            baseline_question = self.form.cleaned_data['baseline_question']
+            follow_up_question = self.form.cleaned_data['follow_up_question']
+            contacts = self.form.cleaned_data['contacts']
+            baseline_minimum = self.form.cleaned_data['baseline_minimum']
+            baseline_maximum = self.form.cleaned_data['baseline_maximum']
+            follow_up_minimum = self.form.cleaned_data['follow_up_minimum']
+            follow_up_maximum = self.form.cleaned_data['follow_up_maximum']
+            start = self.form.cleaned_data['start_date']
+            end = self.form.cleaned_data['end_date']
+
+            # Create a single PollRun for the Baseline Poll
+            self.create_baseline(baseline_question.poll, start, contacts, baseline_question,
+                                 baseline_minimum, baseline_maximum)
+
             # Create a PollRun for each date from start to end dates for the Follow Up Poll
+            loop_count = 1
             for follow_up_date in rrule.rrule(rrule.DAILY, dtstart=start, until=end):
+                if loop_count == 7:  # On the seventh day, add another set of baseline data
```

Comment:
Hmm, if my finger-counting is correct I think this is one-off.

A possible tidy solution: 

```
for loop_count, follow_up_date in enumerate(rrule.rrule(rrule.DAILY, dtstart=start, until=end)):
    if loop_count % 7 == 0:
        ...
```